vortexor/src/vortexor.rs

```diff
@@ -0,0 +1,216 @@
+use {
+    crossbeam_channel::Sender,
+    solana_net_utils::{bind_in_range_with_config, bind_more_with_config, SocketConfig},
+    solana_perf::packet::PacketBatch,
+    solana_sdk::{quic::NotifyKeyUpdate, signature::Keypair},
+    solana_streamer::{
+        nonblocking::quic::DEFAULT_WAIT_FOR_CHUNK_TIMEOUT,
+        quic::{spawn_server_multi, EndpointKeyUpdater, QuicServerParams},
+        streamer::StakedNodes,
+    },
+    std::{
+        net::UdpSocket,
+        sync::{atomic::AtomicBool, Arc, Mutex, RwLock},
+        thread::{self, JoinHandle},
+        time::Duration,
+    },
+};
+
+pub const MAX_QUIC_CONNECTIONS_PER_PEER: usize = 8;
+pub const NUM_QUIC_ENDPOINTS: usize = 8;
+
+pub struct TpuSockets {
+    pub tpu_quic: Vec<UdpSocket>,
+    pub tpu_quic_fwd: Vec<UdpSocket>,
+}
+
+pub struct TpuStreamerConfig {
+    pub tpu_thread_name: &'static str,
+    pub tpu_metrics_name: &'static str,
+    pub tpu_fwd_thread_name: &'static str,
+    pub tpu_fwd_metrics_name: &'static str,
+    pub quic_server_params: QuicServerParams,
+}
+
+pub struct Vortexor {
+    thread_handles: Vec<JoinHandle<()>>,
+    key_update_notifier: Arc<KeyUpdateNotifier>,
+}
+
+struct KeyUpdateNotifier {
+    key_updaters: Mutex<Vec<Arc<EndpointKeyUpdater>>>,
+}
+
+impl KeyUpdateNotifier {
+    fn new(key_updaters: Vec<Arc<EndpointKeyUpdater>>) -> Self {
+        Self {
+            key_updaters: Mutex::new(key_updaters),
+        }
+    }
+}
+
+impl NotifyKeyUpdate for KeyUpdateNotifier {
+    fn update_key(&self, key: &Keypair) -> Result<(), Box<dyn std::error::Error>> {
+        let updaters = self.key_updaters.lock().unwrap();
+        for updater in updaters.iter() {
+            updater.update_key(key)?
+        }
+        Ok(())
+    }
+}
+
+impl Vortexor {
+    pub fn create_tpu_sockets(
```

Comment:
Is this functionally different than what we do in the main validator client? Wondering if the code can be refactored and used here.

Reply:
In the first pr it does is just a thin wrapper of the streamer. I thought about refactoring it and decided against it as  1. I see little code change than wrapping the arguments and sending to streamer.2. The vortexor  will be loaded with other functionality such as key updater interaction, sigverify interaction, forwarder interactions.